src/main/java/seedu/address/logic/commands/room/EditRoomCommand.java

```diff
@@ -131,6 +138,25 @@ private static Room getValidPatient(List<Patient> patientList, Name patientName,
         throw new CommandException(MESSAGE_INVALID_PATIENT_NAME_INPUT);
     }
 
+    /**
+     * Checks if patient is already assigned to a room.
+     *
+     * @param roomList Containing all the rooms.
+     * @param name Of the patient.
+     * @return Boolean value of whether patient is already assigned.
+     */
+    private static boolean isPatientAssignedRoom(List<Room> roomList, Name name) {
```

Comment:
Do you think we should put this function under EditRoomCommand class or should it be the responsibility of RoomList class?

Reply:
Hmm, now thinking about it, it does seem like bad design. I'll change this later and update this pr. I will probably use the observablelist.stream().anyMatch to replace this and put it inside the RoomList.java. Do you think its a good idea?

Reply:
Hmm I was thinking of having the function "isPatientAssignedARoom" in the Model interface instead. Since the Model contains roomList and the patientRecords, Model knowing whether isPatientAssignedARoom kind of makes sense (essentially it will call on another isPatientAssignedARoom method in the roomList class to get the result).  

I think the key here is that the command should only know of Model and should not be accessing roomList directly. (like the Law of Demeter that we learn for this week) That's what I intend to improve for my deletePatientCommand because I also accessed the patient list directly from model, but I think it's not very well designed since there's a deletePatient command directly from model (original code from address book).

Reply:
thats true, let me shift the code there then